### PR TITLE
Add 1.5 billion words Arabic corpus 

### PR DESCRIPTION
**Pull request by zaidalyafeai**
Needs https://github.com/huggingface/datasets/pull/1429 to work.

### CODE REVIEW

**Comment by zaidalyafeai** [comment]
Can't pass dummy data tests. For the instructions, it asks me to generate the following file `dummy_data/Youm7_XML_utf_8.rar/Youm7_utf_8.xml` which is strange, any ideas @lhoestq ?

cc: I tested the data locally and it works, maybe the dummy tests doesn't support `rar` ?

**Comment by lhoestq** [comment]
In the dummy_data.zip files you must include the rar file as if is was already extracted.
In particular here `Youm7_XML_utf_8.rar` is a directory (not an archive).

**Comment by lhoestq** [comment]
Also I'm getting `BadRarFile: Failed the read enough data: req=16384 got=51` while trying to download and extract the `Alittihad_XML_utf_8.rar` file. Do you have this issue as well ?

I have rarfile 4.0

**Comment by zaidalyafeai** [comment]
Sorry it was my mistake, I missed up the directories, it works now. Not sure why you got that error. I have the same version of `rarfile`. Between, there were some suggestions to change the dataset from `1bn_words_arabic` to `arabic_billion_words` like https://github.com/huggingface/datasets/tree/master/datasets/spanish_billion_words.

**Comment by lhoestq** [comment]
I'm ok with renaming the dataset `arabic_billion_words` if you want.
Note that you will need to rename class name `ArabicBillionWords` instead of `BillionWords`
(though `BillionWords` was not matching `1bn_words_arabic` anyway)

You will need to regenerate the dataset_infos.json file after this change.
OR alternatively just replace all mentions of `billion_words` with `arabic_billion_words` in dataset_infos.json <- this trick should save you some time :)

**Comment by lhoestq** [comment]
Hmmm I'm still not able to run it on my side because of the rar error (I'm running macos)
I just tried with rarfile 3.1 and it didn't work either.
I would like to be able to run it end-to-end on my side before merging if you don't mind. Let me investigate this issue a little bit

**Comment by zaidalyafeai** [comment]
No worries, I will investigate it as well.

**Comment by lhoestq** [comment]
Nice thanks, maybe it's just an issue on my side then

**Comment by lhoestq** [comment]
Ok I manage to solve the BadRarFile issue on my side :) 
To fix it I had to install the `unrar` tool for macos (though it seems it's not available with `brew install` anymore, I had to install it from elsewhere).

File: datasets/arabic_billion_words/README.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#@title
```
Comment (lhoestq) [inline datasets/arabic_billion_words/README.md:1]:
```suggestion
```

File: datasets/arabic_billion_words/README.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+## Dataset Structure
+
+### Data Instances
+
+[More Information Needed]
+
+### Data Fields
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/arabic_billion_words/README.md:79]:
```suggestion
The data fields are:
- "url": string, original url of the article,
- "head_line": string, headline of the article,
- "date": string, date of the article,
- "text": string, text content of the article,
```

**Review by lhoestq — APPROVED** [review]
Thanks for adding this one :) 

LGTM

**Pull request merged**